src/clm/commands/write_freq_distribution.py

```diff
@@ -0,0 +1,51 @@
+import argparse
+
+import pandas as pd
+import numpy as np
+from clm.functions import read_file
+
+
+def add_args(parser):
+    parser.add_argument("--sampled_file", type=str, help="Path to the sampled file")
+    parser.add_argument("--test_file", type=str, help="Path to the test file ")
+    parser.add_argument(
+        "--output_file", type=str, help="Path to the save the output file"
+    )
+
+    return parser
+
+
+def write_freq_distribution(sampled_file, test_file, output_file):
+    sampled_data = pd.read_csv(sampled_file)
+    test_smiles = set(read_file(test_file, stream=True, smile_only=True))
+
+    # Label smiles not found in test set as novel
+    sampled_data = sampled_data.assign(
+        is_novel=np.select(
+            [
+                sampled_data["smiles"].isin(test_smiles).values,
```

Comment:
Maybe assign a default value `False` to this column and `True` if this condition is met? (`np.select` may be useful in cases where the conditions are not exhaustive).

Reply:
Simplified the `np.select` by implementing the following two lines: 
```
sampled_data['is_novel'] = True
sampled_data.loc[sampled_data['smiles'].isin(test_smiles), 'is_novel'] = False
```